src/commands/ext-configure.ts

```diff
@@ -48,21 +48,33 @@ export default new Command("ext:configure <extensionInstanceId>")
       const paramSpecWithNewDefaults = paramHelper.getParamsWithCurrentValuesAsDefaults(
         existingInstance
       );
-      const removedLocations = _.remove(paramSpecWithNewDefaults, (param) => {
-        return param.param === "LOCATION";
+      const immutableParams = _.remove(paramSpecWithNewDefaults, (param) => {
+        return param.immutable || param.param === "LOCATION";
+        // TODO: Stop special casing "LOCATION" once all official extensions make it immutable
       });
-      const currentLocation = _.get(existingInstance, "config.params.LOCATION");
+
       const params = await paramHelper.getParams(
         projectId,
         paramSpecWithNewDefaults,
         options.params
       );
-      if (removedLocations.length) {
+      let immutableParamsText = "";
+      for (const immutableParam of immutableParams) {
+        const currentValue = _.get(existingInstance, "config.params." + immutableParam.param);
+        immutableParamsText += `param: ${immutableParam.param}, value: ${currentValue}\n`;
+        params[immutableParam.param] = currentValue;
+      }
+      if (immutableParams.length === 1) {
+        logger.info(
+          `The following param is immutable:\n${immutableParamsText}It cannot be modified. ` +
+            "To set a different value for this param, uninstall this extension, then install a new instance of this extension."
+        );
+      }
+      if (immutableParams.length > 1) {
         logger.info(
-          `Location is currently set to ${currentLocation}. This cannot be modified. ` +
-            `Please uninstall and reinstall this extension to change location.`
+          `The following params are immutable:\n${immutableParamsText}These cannot be modified. ` +
+            "To set different values for these params, uninstall this extension, then install a new instance of this extension."
         );
-        params.LOCATION = currentLocation;
       }
```

Comment:
What would you think of the following? (I tend to not do "build the string" patterns if they can be avoided):

```typescript
const plural = immutableParams.length > 1;
logger.info(`The following param${plural ? "s are" : " is"} immutable:`);
for (const { param } of immutableParams) {
  logger.info(`param: ${param}, value: ${_.get(existingInstance, `config.params.${param}`)}`)
}
// Easier to do one `plural` check here to change the first half of the statement.
logger.info((plural
  ? "To set different values for these params"
  : "To set a different value for this param")
    + ", uninstall the extension, then install a new instance of this extension.");
}
```

This way, the reader isn't trying to track down where newlines are coming from, each param line is generated right in place, it reads top-to-bottom as the logs would appear in the console - I generally would prefer this pattern slightly more. What do you think?

Reply:
The copy looks fine to me.  Y'all's call on the syntax of making the copy display.

Reply:
I can get behind this - Rachel didn't see the newlines in the syntax I had on first look, so its definitely not the most readable atm.